fearless/Modules/Staking/StakingPayoutConfirmation/StakingPayoutConfirmationInteractor.swift

```diff
@@ -84,236 +86,277 @@ final class StakingPayoutConfirmationInteractor {
+    }
 
-        presenter.didRecieve(
-            account: account,
-            rewardAmount: rewardAmount
-        )
+    func handleAccountInfo(result: Result<AccountInfo?, Error>, address _: AccountAddress) {
+        presenter.didReceiveAccountInfo(result: result)
     }
-}
 
-extension StakingPayoutConfirmationInteractor: StakingPayoutConfirmationInteractorInputProtocol {
-    func setup() {
-        subscribeToStashControllerProvider()
-        subscribeToAccountChanges()
-        subscribeToPriceChanges()
-        getRewardData()
+    func handlePayee(result: Result<RewardDestinationArg?, Error>, address _: AccountAddress) {
+        presenter.didReceivePayee(result: result)
     }
+}
 
-    func submitPayout() {
-        guard let closure = createExtrinsicBuilderClosure() else {
-            return
-        }
+// MARK: - SubstrateProviderSubscriber, SubstrateProviderSubscriptionHandler
 
-        presenter.didStartPayout()
+extension StakingPayoutConfirmationInteractor: SubstrateProviderSubscriber,
+    SubstrateProviderSubscriptionHandler {
+    func handleStashItem(result: Result<StashItem?, Error>) {
+        do {
+            let stashItem = try result.get()
 
-        extrinsicService.submit(
-            closure,
-            signer: signer,
-            runningIn: .main
-        ) { [weak self] result in
-            switch result {
-            case let .success(txHash):
-                self?.presenter.didCompletePayout(txHash: txHash)
-            case let .failure(error):
-                self?.presenter.didFailPayout(error: error)
+            clear(dataProvider: &payeeProvider)
+
+            if let stashItem = stashItem {
+                payeeProvider = subscribeToPayeeProvider(
+                    for: stashItem.stash,
+                    runtimeService: runtimeService
+                )
+
+                presenter.didReceiveStashItem(result: .success(stashItem))
+            } else {
+                presenter.didReceiveStashItem(result: .success(nil))
             }
+        } catch {
+            presenter.didReceiveStashItem(result: .failure(error))
         }
     }
+}
 
-    func estimateFee() {
-        guard let closure = createExtrinsicBuilderClosure() else {
-            return
-        }
+// MARK: - RuntimeConstantFetching
 
-        extrinsicService.estimateFee(closure, runningIn: .main) { [weak self] result in
-            switch result {
-            case let .success(info):
-                self?.presenter.didReceive(paymentInfo: info)
-            case let .failure(error):
-                self?.presenter.didReceive(feeError: error)
+extension StakingPayoutConfirmationInteractor: ExtrinsicFeeProxyDelegate {
+    enum FeeType: String {
+        case elaborateBlockWeight
+        case estimateFirstBlock
+        case estimateLastBlock
+    }
+
+    func didReceiveFee(result: Result<RuntimeDispatchInfo, Error>, for identifier: ExtrinsicFeeId) {
+        switch result {
```

Comment:
I think it it more simpler and clear to use ExtrinsicOperationFactory directly for fee calculation because one can benifit operation chaining. You can even chain it with constant extraction from runtime.